src/transactionIntentsFactories/delegationTransactionIntentsFactory.spec.ts

```diff
@@ -0,0 +1,301 @@
+import { DelegationTransactionIntentsFactory } from "./delegationTransactionIntentsFactory";
+import { assert } from "chai";
+import { DELEGATION_MANAGER_SC_ADDRESS } from "../constants";
+import { ValidatorPublicKey } from "@multiversx/sdk-wallet-next";
+import { SignableMessage } from "../signableMessage";
+
+describe("test delegation intents factory", function () {
+    const delegationFactory = new DelegationTransactionIntentsFactory({
+        chainID: "D",
+        minGasLimit: 50000,
+        gasLimitPerByte: 1500,
+        gasLimitStake: 5000000,
+        gasLimitUnstake: 5000000,
+        gasLimitUnbond: 5000000,
+        gasLimitCreateDelegationContract: 50000000,
+        gasLimitDelegationOperations: 1000000,
+        additionalGasLimitPerValidatorNode: 6000000,
+        additionalGasLimitForDelegationOperations: 10000000
+    });
+
+    it("should build intent for new delegation contract", async function () {
+        const sender = Address.fromBech32("erd18s6a06ktr2v6fgxv4ffhauxvptssnaqlds45qgsrucemlwc8rawq553rt2");
+        const delagationCap = new BigNumber("5000000000000000000000");
+        const serviceFee = new BigNumber(10);
+        const value = new BigNumber("1250000000000000000000");
+
+        const intent = delegationFactory.createTransactionIntentForNewDelegationContract({
+            sender: sender,
+            totalDelegationCap: delagationCap,
+            serviceFee: serviceFee,
+            value: value
+        });
+
+        assert.equal(intent.sender, "erd18s6a06ktr2v6fgxv4ffhauxvptssnaqlds45qgsrucemlwc8rawq553rt2");
+        assert.equal(intent.receiver, DELEGATION_MANAGER_SC_ADDRESS);
+        assert.isDefined(intent.data);
+        assert.deepEqual(intent.data, Buffer.from("createNewDelegationContract@010f0cf064dd59200000@0a"));
+        assert.equal(intent.gasLimit.valueOf(), 60126500);
+        assert.equal(intent.value, value);
+    });
+
+    it("should build intent for adding nodes", async function () {
+        const sender = Address.fromBech32("erd18s6a06ktr2v6fgxv4ffhauxvptssnaqlds45qgsrucemlwc8rawq553rt2");
+        const delegationContract = Address.fromBech32("erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqtllllls002zgc");
+        const publicKey = new ValidatorPublicKey(Buffer.from("e7beaa95b3877f47348df4dd1cb578a4f7cabf7a20bfeefe5cdd263878ff132b765e04fef6f40c93512b666c47ed7719b8902f6c922c04247989b7137e837cc81a62e54712471c97a2ddab75aa9c2f58f813ed4c0fa722bde0ab718bff382208", "hex"));
+
+        const signedMessage = new SignableMessage();
+        signedMessage.applySignature(Buffer.from("81109fa1c8d3dc7b6c2d6e65206cc0bc1a83c9b2d1eb91a601d66ad32def430827d5eb52917bd2b0d04ce195738db216", "hex"));
+
+        const intent = delegationFactory.createTransactionIntentForAddingNodes({
+            sender: sender,
+            delegationContract: delegationContract,
+            publicKeys: [publicKey],
+            signedMessages: [signedMessage.getSignature()]
```

Comment:
Since it's a `mock message`, we can do:

```
const mockMessage = {
    getSignature: () => Buffer.from("...", hex)
}
```

Should work. Then, we'll remove `new SignableMessage()` and `applySignature()` from the test.

Reply:
Done